### PR TITLE
[Fixes #9088] Store eip1581 path

### LINKED ISSUE #9088: Store eip1581 extended key
# Problem

After a discussion triggered by https://github.com/status-im/status-react/pull/9033 it was decided to store the extended key at path `m/43'/60'/1581'` so that in the future if and when we would like to derive an encryption key from this path it will be available to the user without having to input their seed phrase again.
This has no impact on the user experience, so it's just a technical task.

## Acceptance Criteria

- The key at path `m/43'/60'/1581'` should be stored encrypted on disk, similarly as we do with the chat key (`m/43'/60'/1581/0'/0'`) and wallet key (`m/44'/60'/0'/0/0`)
- Stop storing the master key
- Store instead of the master key the key at `m/44'/60'/0'/0/` and use that to derive wallets
- Ensure that this key can be accessed by the application
- Update https://notes.status.im/QxNC1H2cQMG7t8GvEb_Ctw with the relevant key informations

### PR DESCRIPTION
**Pull request by cammellos**
When creating the account we store as well the path specified in eip1581
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1581.md ,
`m / 43' / 60' / 1581'`.

The reason for doing so is that eventually we might want to derive an
encryption key from it, which would require the user to re-enter their
seed phrase if we would not store this.

This commit changes also the behavior **not to** store the master key, and instead store the key at 
`m /44'/60' /0'/0` from which wallets can be derived.

I could not test on keycard as I do not own one.

status: ready

### CODE REVIEW

**Review by gravityblast — APPROVED** [review]
LGTM

**Comment by cammellos** [comment]
@guylouis @dmitryn I am not able to test this change on keycard accounts as I don't own one.
The relevant changes is that the master key is not stored anymore on the filesystem for normal accounts (as I understand with a keycard this has never been the case, the master key is indeed stored, but only on the keycard).
For non-keycard accounts we store `m /44'/60' /0'/0 ` and use that to create new wallet accounts (which the keycard is not storing), and if that has not been created, we use the address of the master key instead (which will fail for non keycard accounts, but should work for keycards, as my understanding is that we do store the master key there).

Is that correct or I misunderstand something?

@churik are you able to test creating/recovering an account and generating new wallets (in both cases) on this?

**Comment by churik** [comment]
OFC, on it!

**Comment by corpetty** [comment]
It seems as though this was prematurely approved without testing out keycard.  Am I misreading that?  If that is the case, maybe a label like `needs testing` would be helpful as `status: ready` signals that nothing needs to be done even though it's intent was more along the lines of "the code is done."

**Comment by churik** [comment]
Tested on Android 8:
- creating ordinary account (passcode with special chars) 
- restoring ordinary account (passcode with special chars) 
- creating account with keycard
- restoring account with keycard (create account, reinstall, restore)

All cases work fine for me.

**Comment by churik** [comment]
@corpetty  we have such label `request-manual-qa`

**Comment by cammellos** [comment]
@corpetty I think there's some confusion on the workflow:

`status: ready` means that the PR is ready to be reviewed (as opposed to status: wip which means that we haven't finished writing the code yet), not that is ready to be merged or that it has been tested (we add a label/move in the to merge review).

In terms of testing, up until last week all PRs went through manual QA, and I believe we add a label when manual QA is required. This one does not really need manual QA, other than someone needs to validate that it works with keycard, as specified in the comment (any developer can do).

@churik thanks, I'll wait for either @yenda or @flexsurfer to approve then is ready to merge.

**Comment by yenda** [comment]
Does it means that a second account created with keycard will be different than one created with the same multiaccount without keycard?

**Comment by cammellos** [comment]
@yenda No it should not, the path is identical,  `m /44'/60' /0'/0/x`, the only difference is that in the case of keycard it will be generated from the master key, while if it's not a keycard account, it will be generated from the `BIP44 Wallet Root key`.

But best to check that is the case:

@churik would you mind doing a test please if you have some time?

1) Import the same account once with keycard once without
2) Create an extra wallet on both, they should have identical addresses (the initial one as well)

Thanks!

**Comment by cammellos** [comment]
mmh turns out that they do seem to differ (I tried with a pre this commit and this commit, not on keycard), it might be that the path is relative to the key, let me check

**Comment by cammellos** [comment]
@yenda turns out we need to use the relative path, so if deriving from `m/a/b/c` you derive `m/0` for example, but if deriving from the master key, you'd derive `m/a/b/c/0`, which makes sense.

@churik could you have a look at the test with keycard when you have some time? It should show identical wallet addresses now.
Thanks!

**Comment by churik** [comment]
@cammellos 

> Import the same account once with keycard once without

It shouldn't be possible according to https://github.com/status-im/status-react/issues/9062#issuecomment-537965163

> Create an extra wallet on both, they should have identical addresses (the initial one as well)

Due to https://github.com/status-im/status-react/issues/9019 it is not possible for now

So can't help here.

**Comment by guylouis** [comment]
@cammellos 
just want to make sure: are we blocked here by something ?
can keycard team help in anyway ? cc @dmitryn

**Comment by cammellos** [comment]
@guylouis No, I think it's fine, as one functionality is not supported (restore same account with keycard), and the other is not implemented yet (add multiple wallets with keycard).
Just waiting for a review from a clojure dev, otherwise is ready to merge. cc @flexsurfer @yenda

File: src/status_im/constants.cljs
Comment (flexsurfer) [inline src/status_im/constants.cljs:245]: before "m/44'/60'/0'/0/0" after "m/44'/60'/0'/0"/0" is it the same path ?

Comment (flexsurfer) [reply src/status_im/constants.cljs:245]: for some reason before it was "0/"

Comment (flexsurfer) [reply src/status_im/constants.cljs:245]: was it mistake?

Comment (cammellos) [reply src/status_im/constants.cljs:245]: wait, not sure i understand, path looks the same to me:

before:

m/44'/60'/0'/0
m/44'/60'/0'/0/0

after:

m/44'/60'/0'/0
(str "m/44'/60'/0'/0" "/0")

looks the same to me, or i am missing something?

Comment (flexsurfer) [reply src/status_im/constants.cljs:245]: oh my, confused by "

**Comment by flexsurfer** [comment]
@cammellos could you summarize why so many changes? I thought we just want to store an encryption key in this PR?

File: src/status_im/wallet/accounts/core.cljs
Comment (flexsurfer) [inline src/status_im/wallet/accounts/core.cljs]: just wondering why keys in a column and not in a row?

Comment (cammellos) [reply src/status_im/wallet/accounts/core.cljs]: i like columns :) but i can put them on a row, np

**Comment by cammellos** [comment]
@flexsurfer that was he initial idea, but turns out we were also storing the master key on disk, which we should not have ( after a discussion with @corpetty), so I have changed the behaviour to store the root wallet key instead. It is explained as well in the issue

**Review by flexsurfer — APPROVED** [review]

**Pull request merged**
